device-to-group-command.py

```diff
@@ -0,0 +1,404 @@
+'''
+esper app usage
+
+usage: esper_group_actions [-h]
+                            -c  {uninstall, whitelist, brightness, alarm_volume, ring_volume, music_volume, notification_volume, bluetooth, wifi, gps, ping, reboot}
+                            -g  GROUP_ID
+                            -v  VALUE
+
+example:
+
+./esper_group_actions -g 52ecfb3c-d1ad-4e66-8cf9-85daff8d7f3c -c ring_volume -v 50
+./esper_group_actions -g 52ecfb3c-d1ad-4e66-8cf9-85daff8d7f3c -c ping
+
+Runs on all active devices in a group. API key is required.
+'''
+
+import sys
+import requests
+import esperclient
+import time
+from esperclient.rest import ApiException
+
+import argparse
+active_device_list = []
+inactive_device_list = []
+
+#### TYRELL CORP CONFIG #######
+
+configuration = esperclient.Configuration()
+configuration.host = 'https://tyrellcorp-api.shoonyacloud.com/api'
+configuration.api_key['Authorization'] = 'RaCh1Ty3L181AdE4uN32e593rD3ck36'
+configuration.api_key_prefix['Authorization'] = 'Bearer'
+enterprise_id = '5da4cf47-398c-4cea-82be-2796a3f23c3c'
+
+###############################
+
+def run_whitelist_command(device, package_name):
+    api_instance = esperclient.CommandsApi(esperclient.ApiClient(configuration))
+    command_args = esperclient.CommandArgs(package_name=package_name)
+    command = esperclient.CommandRequest(command='ADD_TO_WHITELIST', command_args=command_args)
+    try:
+        api_response = api_instance.run_command(enterprise_id, device.id, command)
+
+        if api_response.id is not None:
+            print('Whitelist command fired successfully on ' +  device.device_name + ' for ' + package_name)
+    except ApiException as e:
+        print("Exception when calling CommandsApi->run_command: %s\n" % e)
+
+def get_whitelist_api_url(device_id):
+    return configuration.host + '/enterprise/' + enterprise_id + '/device/' + device_id + '/app/'
+
+def is_package_whitelisted(device_id, package_name):
+    url = get_whitelist_api_url(device_id)
+    querystring = {"package_name": package_name, "whitelisted":"false"}
+    payload = ""
+    headers = {
+        'Authorization': "Bearer " + configuration.api_key['Authorization'],
+        'cache-control': "no-cache",
+        }
+
+    response = requests.request("GET", url, data=payload, headers=headers, params=querystring)    
+    print("Whitelist query match: " + str(response.json().get('count')))
+    return False if response.json().get('count') == 1 else True
+
+
+def whitelist_package_in_group_devices(device_list, package_name):
+    # Fetch all devices in group
+
+    # None check
```

Comment:
you might want to remove these comments.  As these operation are not present here.

Reply:
done